src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py

```diff
@@ -36,17 +63,35 @@ class PhEDExInjectorPoller(BaseWorkerThread):
 
     Poll the DBSBuffer database and inject files as they are created.
     """
-    def __init__(self, config, phedex, nodeMappings):
+    def __init__(self, config):
         """
         ___init___
 
         Initialise class members
         """
         BaseWorkerThread.__init__(self)
-        self.config = config
-        self.phedex = phedex
         self.dbsUrl = config.DBSInterface.globalDBSUrl
 
+        self.pollCounter = 0
+        self.subFrequency = None
+        if getattr(config.PhEDExInjector, "subscribeDatasets", False):
+            pollInterval = config.PhEDExInjector.pollInterval
+            subInterval = config.PhEDExInjector.subscribeInterval
+            self.subFrequency = max(1, int(round(subInterval/pollInterval)))
```

Comment:
I didn't get why we need another frequency/polling yet. Why don't we simply use the configuration subscribeInterval for that cycle of subscribe+delete?